### ISSUE TITLE
Can't pass Rule MD007

### ISSUE THREAD

**Issue by chientrm** [markdownlint/markdownlint#423]
## Description

My md:

```md
### Enable required APIs

1. Browse to [Google Cloud Console](https://console.cloud.google.com/).
1. Obtains Project ID
   ![Project Id](./project-id.png)
1. From `Navigation Menu` -> `APIs & Services` -> `Enabled APIs & Services`.
1. Click button `ENABLE APIS AND SERVICES`.
1. Search for these APIs and enable all of them:
   - Cloud Build API
   - Compute Engine API
   - Service Management API
   - Google Cloud Storage JSON API
```

### Environment

**MDL Version** Latest version install via gem on Github Action `ubuntu-latest`

### Expected Behavior

Rule passed

### Actual Behavior

```
MD007 Unordered list indentation
```

**Comment by chientrm** [comment]
For anyone looking, the solution takes you 5 spaces to get it right.

```md
### Enable required APIs

1. Browse to [Google Cloud Console](https://console.cloud.google.com/).
1. Obtains Project ID
   ![Project Id](./project-id.png)
1. From `Navigation Menu` -> `APIs & Services` -> `Enabled APIs & Services`.
1. Click button `ENABLE APIS AND SERVICES`.
1. Search for these APIs and enable all of them:
     - Cloud Build API
     - Compute Engine API
     - Service Management API
     - Google Cloud Storage JSON API
```

**Comment by jaymzh** [comment]
I cannot reproduce this at least on `master`:

```
[phil@rider RVM(2.7.1) ~]$ cat /tmp/test.md 
# Header

## Some Section

### Enable required APIs

1. Browse to [Google Cloud Console](https://console.cloud.google.com/).
1. Obtains Project ID
   ![Project Id](./project-id.png)
1. From `Navigation Menu` -> `APIs & Services` -> `Enabled APIs & Services`.
1. Click button `ENABLE APIS AND SERVICES`.
1. Search for these APIs and enable all of them:
   - Cloud Build API
   - Compute Engine API
   - Service Management API
   - Google Cloud Storage JSON API
[phil@rider RVM(2.7.1) ~]$ mdl /tmp/test.md 
[phil@rider RVM(2.7.1) ~]$ 
```
